File: Algorithm/code/exp-1/1. 统计数字问题（P8）/C/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main()
{
    FILE *fp;
    int n;
    if((fp=fopen("input.txt", "r"))==NULL)
    {
        printf("Error: txt file can't open!\n");
        exit(1);
    } 
    else 
    {
        while(fscanf(fp,"%d",&n) != EOF);
    }

    char str[10];
    int result[10] = {0};
    for(int i=1; i<=n; i++)
    {
        sprintf(str, "%d", i);
        for(int j=0; str[j]; j++)
            result[str[j]-'0']++;
    }

    if((fp=fopen("output.txt", "w"))==NULL)
    {
        printf("Error: txt file can't open!\n");
        exit(1);
    } 
    for(int i=0; i<10; i++)
        fprintf(fp, "%d\n", result[i]);
    fclose(fp);

    return 0;
}

File: Algorithm/code/exp-1/1. 统计数字问题（P8）/Python/main.py
# 文件读取
fp = open('input.txt', 'r')
page = int(fp.read())
fp.close()

# 结果存放数组
result = [0]*10

# 遍历每个页号
for i in range(1, page+1):
    stri = str(i)
    for char in stri:
        index = int(char)
        result[index] += 1

# 打印输出
fp = open('output.txt', 'w')
for num in result:
    fp.write(str(num)+'\n')
    print(num)
fp.close()